### PR TITLE
Dragonflight

### PR DESCRIPTION
**Pull request by p3lim**
This should contain most, if not all of the fixes and features for Dragonflight.
We'll also merge in any other branches from the [11 milestone](https://github.com/oUF-wow/oUF/milestone/12) if time allows it.

### CODE REVIEW

File: elements/auras.lua [hunk 1]
```diff
@@ -422,17 +508,96 @@ local function UpdateAuras(self, event, unit)
...
+				end
+
+				table.sort(buffs.sorted, buffs.SortBuffs or buffs.SortAuras or SortAuras)
+			end
+		end
 
-		local fromRange, toRange
-		if(buffs.PreSetPosition) then
-			fromRange, toRange = buffs:PreSetPosition(numBuffs)
+		if(not buffsChanged) then return end
```
Comment (brendanblackwood) [inline elements/auras.lua:589]: I believe this is causing `UpdateAuras` to exit early if there have been no buff changes even if there have been debuff changes, resulting in debuffs not being added or removed.

Comment (ls-) [reply elements/auras.lua:589]: Ehm, it's in `.Buffs`, so obv it only handles buffs...

Comment (brendanblackwood) [reply elements/auras.lua:589]: I could totally be misunderstanding how it's supposed to be used, but it appears that if a layout has set both `.Buffs` and `.Debuffs`, it relies on `UpdateAuras` to update both of them, not just the buffs. In this case, we are returning from `UpdateAuras` if there are no changes to buffs even if there are changes to debuffs.

Comment (ls-) [reply elements/auras.lua:589]: Ah, true >_> Good catch. More work 😒🔫

**Comment by p3lim** [comment]
TODO (according to @ls-):
- [ ] castbars needs some tweaking
- [ ] docs needs updating
- [ ] merge #592
- [ ] more testing (we should have an example layout that covers everything tbh)

**Comment by ls-** [comment]
Code-wise I think we're ready. Just to clarify on a couple of things.

- I chose against implementing cooldowns for the `Essence` classpower. The way Blizz implemented them is honestly wack. so overcomplicating the element to support that wasn't worth it. If you want them, use `PostUpdate` and `GetPowerRegenForPowerType` to show the "cooldown".

- Castbar empowerment stages. We only add pips to the element, if you need fancy colouring like in the default UI, use those pips as anchors, that's what the default UI does as well. Both horizontal and vertical bars are supported, the reverse fill is supported as well.

- Despite how complex `auras` changes might look, the way the element works now is pretty straightforward. Also, its performance isn't an issue, on average (full and partial updates combined) the new implementation is 30% faster than what we had previously, and even that wasn't slow either. For instance, here's the avg performance per call after 1k calls with like 24-32 active auras:
image

So what's left is to update the documentation where it's outdated or missing.

**Comment by p3lim** [comment]
Posted upgrade guide: https://www.wowinterface.com/forums/showthread.php?t=59273

**Review by ls- — DISMISSED** [review]
👍🏼

**Review by ls- — APPROVED** [review]
👍🏼

**Pull request merged**